unit-tests/src/test/java/org/eclipse/collections/impl/list/mutable/primitive/BoxedMutableBooleanListTest.java

```diff
@@ -0,0 +1,185 @@
+        BoxedMutableBooleanList list = this.classUnderTest();
+        Verify.assertSize(2, list);
+        list.add(false);
+        Verify.assertSize(3, list);
+    }
+
+    @Test
+    public void add()
+    {
+        BoxedMutableBooleanList list = this.classUnderTest();
+        list.add(Boolean.FALSE);
+        Assert.assertEquals(new BoxedMutableBooleanList(new BooleanArrayList(true, true, false)), list);
+        list.add(Boolean.TRUE);
+        Assert.assertEquals(new BoxedMutableBooleanList(new BooleanArrayList(true, true, false, true)), list);
+    }
+
+    @Test
+    public void mutationOfOriginalList()
+    {
+        BooleanArrayList originalList = new BooleanArrayList(true, false);
+        BoxedMutableBooleanList list = new BoxedMutableBooleanList(originalList);
+        Assert.assertEquals(list, Lists.mutable.of(Boolean.TRUE, Boolean.FALSE));
+
+        originalList.add(true);
+        Assert.assertEquals(list, Lists.mutable.of(Boolean.TRUE, Boolean.FALSE, Boolean.TRUE));
+
+        originalList.remove(true);
+        Assert.assertEquals(list, Lists.mutable.of(Boolean.FALSE, Boolean.TRUE));
+
+        originalList.addAllAtIndex(1, false);
+        Assert.assertEquals(list, Lists.mutable.of(Boolean.FALSE, Boolean.FALSE, Boolean.TRUE));
+
+        originalList.clear();
+        Verify.assertEmpty(list);
+    }
+
+    @Test
+    public void addAll()
+    {
+        BoxedMutableBooleanList list = this.classUnderTest();
+        list.addAll(0, Lists.mutable.of(Boolean.FALSE, Boolean.FALSE));
+        Assert.assertEquals(new BoxedMutableBooleanList(new BooleanArrayList(false, false, true, true)), list);
+
+        list.addAll(4, Lists.mutable.of(Boolean.FALSE, Boolean.TRUE));
+        Assert.assertEquals(new BoxedMutableBooleanList(new BooleanArrayList(false, false, true, true, false, true)), list);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void addAllIndexOutOfBounds()
+    {
+        this.classUnderTest().addAll(5, Lists.mutable.of(true));
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void addAllNegativeIndex()
+    {
+        this.classUnderTest().addAll(-2, Lists.mutable.of(true));
+    }
+
+    @Test
+    public void clear()
+    {
+        BoxedMutableBooleanList list = this.classUnderTest();
+        list.clear();
+        Verify.assertEmpty(list);
+    }
+
+    @Test
+    public void get()
+    {
+        BoxedMutableBooleanList list = this.classUnderTest();
+        list.add(Boolean.FALSE);
+        Assert.assertTrue(list.get(0));
+        Assert.assertFalse(list.get(2));
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void getIndexOutOfBounds()
+    {
+        this.classUnderTest().get(2);
+    }
+
+    @Test
+    public void set()
+    {
+        BoxedMutableBooleanList list = this.classUnderTest();
+        list.set(0, Boolean.FALSE);
+        Assert.assertEquals(new BoxedMutableBooleanList(new BooleanArrayList(false, true)), list);
```

Comment:
I think this should be a JDK `ArrayList` or EC `FastList` of `Boolean` to make sure the equals contract works correctly.

Reply:
Fixed. Thanks.